loki/transformations/tests/test_inline.py

```diff
@@ -164,44 +164,89 @@ def test_transform_inline_elemental_functions_extended(tmp_path, builder, fronte
   v2 = multiply(v1, 6._real64) + multiply_single_line(v1, 3._real64)
   v3 = add(param1, 200._real64) + add(150._real64, 150._real64) + multiply(6._real64, 11._real64)
 end subroutine transform_inline_elemental_functions_extended
+"""
+
+    fcode_arr = """
+subroutine transform_inline_elemental_functions_extended_array(v1, v2, v3, len)
+  use iso_fortran_env, only: real64
+  use multiply_extended_mod, only: multiply, multiply_single_line, add
+  integer, intent(in) :: len  
+  real(kind=real64), intent(in) :: v1(len)
+  real(kind=real64), intent(inout) :: v2(len), v3(len)
+  real(kind=real64), parameter :: param1 = 100.
+  integer, parameter :: arr_index = 1
+
+  v2 = multiply(v1(:), 6._real64) + multiply_single_line(v1(:), 3._real64)
+  v3 = add(param1, 200._real64) + add(v1, 150._real64) + multiply(v1(arr_index), v2(1))
+end subroutine transform_inline_elemental_functions_extended_array
 """
```

Comment:
Let's make the array case a separate test, please.

You can still avoid replicating the module code etc using the following pattern:

```py
@pytest.fixture(name='multiply_extended_mod', params=available_frontends())
def fixture_multiply_extended_mod(request, tmp_path):
    fcode_module = """
...
    """

    frontend = request.param
    module = Module.from_source(fcode_module, frontend=frontend, xmods=[tmp_path])
    return module, frontend

def test_transform_inline_elemental_functions_extended_scalar(multiply_extended_mod, tmp_path):
    module, frontend = multiply_extended_mod

    fcode = "..."

    routine = Subroutine.from_source(fcode, frontend=frontend, definitions=[module], xmods=[tmp_path])
    ...

def test_transform_inline_elemental_functions_extended_arr(multiply_extended_mod, tmp_path):
    module, frontend = multiply_extended_mod

    fcode_arr = "..."

    routine = Subroutine.from_source(fcode_arr, frontend=frontend, definitions=[module], xmods=[tmp_path])
    ...

```